src/components/base/InternalTextInput/InternalTextInput.tsx

```diff
@@ -43,33 +51,52 @@ const StyledInput = styled.input`
   ::-webkit-search-results-decoration {
     appearance: none;
   }
+
+  ${spacingStyle}
+  ${sizeStyle}
 `;
 
 export type InternalTextInputProps = StyledInputProps & {
-  onFocus?: (e: FocusEvent<HTMLInputElement>) => void;
-  onBlur?: (e: FocusEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  /**
+   * Fired only when the input is focused for the first time
+   */
   onInitialFocus?: (e: FocusEvent<HTMLInputElement>) => void;
 };
 
+/**
+ *
+ * An unstyled input to be used as a basis for UI input components.
+ * Supports all the props of a regular `HTMLInputElement`
+ *
+ * 🚨 The CSS `outline` is disabled so a focus ring must be applied by the consuming component.
+ *
+ * The following callbacks are available for tracking focus events:
+ *
+ *  ### onFocus
+ * `(e: FocusEvent<HTMLInputElement>) => void;`
+ *  ### onBlur
+ * `(e: FocusEvent<HTMLInputElement>) => void;`
+ *  ### onInitialFocus
+ * `(e: FocusEvent<HTMLInputElement>) => void;`<br>
+ *  occurs only when the input is focused for the first time
+ *
+ */
 export const InternalTextInput = (props: InternalTextInputProps) => {
-  const { onInitialFocus, onBlur, onFocus, ...rest } = props;
+  const { onInitialFocus, onFocus, ...rest } = props;
 
   const hadInitialFocused = useRef(false);
 
   const handleOnFocus = (e: FocusEvent<HTMLInputElement>) => {
     if (!hadInitialFocused.current && props.onInitialFocus) {
       props.onInitialFocus(e);
       hadInitialFocused.current = true;
-    } else if (props.onFocus) {
-      props.onFocus(e);
     }
-  };
 
-  const handleOnBlur = (e: FocusEvent<HTMLInputElement>) => {
```

Comment:
why remove the blur event handler ?

Reply:
The code was redundant since the prop is forwarded to the underlying input in the rest arg